test/transformations/LandOrPropertyTransformerSpec.scala

```diff
@@ -119,6 +119,113 @@ class LandOrPropertyTransformerSpec extends AnyFreeSpec with Matchers with Optio
         )
       )
     }
+
+    "should not include lease details when leased is false" in {
+
+      val userAnswers = emptyUserAnswers
+        .unsafeSet(LandPropertyInUKPage(srn, index1of5000), true)
+        .unsafeSet(LandOrPropertyPostcodeLookupPage(srn, index1of5000), postcodeLookup)
+        .unsafeSet(AddressLookupResultsPage(srn, index1of5000), List(address, address, address))
+        .unsafeSet(LandOrPropertyChosenAddressPage(srn, index1of5000), address)
+        .unsafeSet(
+          LandRegistryTitleNumberPage(srn, index1of5000),
+          ConditionalYesNo.yes[String, String]("landRegistryTitleNumber")
+        )
+        .unsafeSet(WhyDoesSchemeHoldLandPropertyPage(srn, index1of5000), SchemeHoldLandProperty.Transfer)
+        .unsafeSet(LandOrPropertyTotalCostPage(srn, index1of5000), Money(100000.0))
+        .unsafeSet(IsLandOrPropertyResidentialPage(srn, index1of5000), false)
+        .unsafeSet(IsLandPropertyLeasedPage(srn, index1of5000), false)
+        .unsafeSet(LandOrPropertyTotalIncomePage(srn, index1of5000), Money(100000.0))
+
+      val request = DataRequest(allowedAccessRequest, userAnswers)
+
+      val result = transformer.transformToEtmp(srn, Some(true), userAnswers)(request)
+      result mustBe Some(
+        LandOrProperty(
+          recordVersion = None,
+          optLandOrPropertyHeld = Some(true),
+          optDisposeAnyLandOrProperty = Some(false),
+          landOrPropertyTransactions = List(
+            LandOrPropertyTransactions(
+              prePopulated = None,
+              propertyDetails = PropertyDetails(
+                landOrPropertyInUK = true,
+                addressDetails = address,
+                landRegistryTitleNumberKey = true,
+                landRegistryTitleNumberValue = "landRegistryTitleNumber"
+              ),
+              heldPropertyTransaction = HeldPropertyTransaction(
+                methodOfHolding = SchemeHoldLandProperty.Transfer,
+                dateOfAcquisitionOrContribution = None,
+                optPropertyAcquiredFromName = None,
+                optPropertyAcquiredFrom = None,
+                optConnectedPartyStatus = None,
+                totalCostOfLandOrProperty = 100000.0,
+                optIndepValuationSupport = None,
+                optIsLandOrPropertyResidential = Some(false),
+                optLeaseDetails = None,
+                optLandOrPropertyLeased = Some(false),
+                optTotalIncomeOrReceipts = Some(100000.0)
+              ),
+              optDisposedPropertyTransaction = None
+            )
+          )
+        )
+      )
+    }
+
+    "should not include lease details when leased is not set" in {
+
+      val userAnswers = emptyUserAnswers
```

Comment:
Duplication could be reduced if you define this as a member variable on the top level and use that in both test, copying it with additional unsafeSets where needed. Example on line 107 above.

Reply:
done